src/hw_isolation_event/hw_status_manager.cpp

```diff
@@ -298,6 +339,100 @@ void Manager::restoreHardwaresStatusEvent()
         });
 }
 
+void Manager::onHostStateChange(sdbusplus::message::message& message)
+{
+    dbus_type::Interface interface;
+    dbus_type::Properties properties;
+
+    try
+    {
+        message.read(interface, properties);
+
+        for (const auto& property : properties)
+        {
+            if (property.first == "CurrentHostState")
+            {
+                if (const auto* propVal =
+                        std::get_if<std::string>(&property.second))
+                {
+                    if (*propVal ==
+                        "xyz.openbmc_project.State.Host.HostState.Quiesced")
+                    {
+                        restoreHardwaresStatusEvent();
+                    }
+                }
+                else
+                {
+                    log<level::ERR>(
```

Comment:
File scope: Revisit error log strategy , journal log  is good to debug only short window. Need error log incase for any specific error condition , which is blocking the functionality.